aries_cloudagent/messaging/jsonld/routes.py

```diff
@@ -45,31 +49,33 @@ async def sign(request: web.BaseRequest):
     response = {}
     try:
         context: AdminRequestContext = request["context"]
+        session = await context.session()
+        try:
+            session.inject(BaseWallet, required=True)
+        except Exception:
+            raise web.HTTPForbidden(reason="No wallet available")
         body = await request.json()
         verkey = body.get("verkey")
         doc = body.get("doc")
-        credential = doc["credential"]
-        signature_options = doc["options"]
-
-        async with context.session() as session:
-            wallet = session.inject(BaseWallet, required=False)
-            if not wallet:
-                raise web.HTTPForbidden(reason="No wallet available")
-            document_with_proof = await sign_credential(
-                credential, signature_options, verkey, wallet
-            )
-
-        response["signed_doc"] = document_with_proof
-    except (DroppedAttributeError, MissingVerificationMethodError) as err:
-        response["error"] = str(err)
-
+        doc_with_proof = await sign_credential(
+            session, doc.get("credential"), doc.get("options"), verkey
+        )
+        response["signed_doc"] = doc_with_proof
+    except (WalletError, DroppedAttributeError, MissingVerificationMethodError) as err:
+        response["error"] = repr(err)
     return web.json_response(response)
 
 
 class VerifyRequestSchema(OpenAPISchema):
     """Request schema for signing a jsonld doc."""
 
-    verkey = fields.Str(required=True, description="verkey to use for doc verification")
+    verkey = fields.Str(
+        required=False, description="verkey to use for doc verification"
+    )
+    verification_method = fields.Str(
+        required=False,
+        description="DID URL to the Verification Method to use to verify doc",
+    )
```

Comment:
Dead code?

Reply:
verification_method is dead code. I will remove it.